tools/wpt/android.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import os
+import platform
+import shutil
+import subprocess
+
+import requests
+from mozrunner.devices import android_device
+
+android_device.TOOLTOOL_PATH = os.path.join(os.path.dirname(__file__),
+                                            os.pardir,
+                                            "third_party",
+                                            "tooltool",
+                                            "tooltool.py")
+
+here = os.path.abspath(os.path.dirname(__file__))
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+
+def get_parser_install():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--reinstall", action="store_true", default=False,
+                        help="Force reinstall even if the emulator already exists")
+    return parser
+
+
+def get_parser_start():
+    return get_parser_install()
+
+
+def get_sdk_path(dest):
+    if dest is None:
+        # os.getcwd() doesn't include the venv path
+        dest = os.path.join(wpt_root, "_venv")
+    dest = os.path.join(dest, 'android-sdk')
+    return os.path.abspath(os.environ.get('ANDROID_SDK_PATH', dest))
+
+
+def uninstall_sdk(dest=None):
+    path = get_sdk_path(dest)
+    if os.path.exists(path) and os.path.isdir(path):
+        shutil.rmtree(path)
+
+
+def install_sdk(logger, dest=None):
+    sdk_path = get_sdk_path(dest)
+    if os.path.isdir(sdk_path):
+        logger.info("Using SDK installed at %s" % sdk_path)
+        return sdk_path, False
+
+    if not os.path.exists(sdk_path):
+        os.makedirs(sdk_path)
+
+    os_name = platform.system().lower()
+    if os_name not in ["darwin", "linux", "windows"]:
+        logger.error("Unsupported platform %s" % os_name)
+        raise NotImplementedError
+
+    os_name = 'darwin' if os_name == 'macosx' else os_name
+    url = 'https://dl.google.com/android/repository/sdk-tools-%s-4333796.zip' % (os_name,)
```

Comment:
Is there a fixed path for the latest SDK? If we want to pin the SDK version, I'd prefer to do that in a manifest/config file.

Reply:
I just copied https://searchfox.org/mozilla-central/source/python/mozboot/mozboot/android.py#186 I assume that's pinned to something and that pinning has a reason, but I don't know what it is. I agree that we should make this more generic somehow, but idk how advantageous a config file is. What's the use case you have in mind?

Reply:
Not at the moment, but based on my limited Android dev experience, SDKs are generally backward compatible (and you can select a wide range of AVD versions from the AVD manager despite the version of SDK itself), so I'd hope the latest version would just work.

Anyway, it doesn't matter too much at the moment; I'd be happy to see a TODO here and leave this as is.